tests/parser/ACTIONX.cpp

```diff
@@ -1056,3 +1056,52 @@ TSTEP
     }
 
 }
+
+BOOST_AUTO_TEST_CASE(Action_WELPI) {
+    const auto deck_string = std::string{ R"(
+SCHEDULE
+
+
+WELSPECS
+    'PROD1' 'G1'  1 1 10 'OIL' /
+/
+
+COMPDAT
+ 'PROD1'  1  1   1   1 'OPEN' 1*   32.948   0.311  3047.839 1*  1*  'X'  22.100 /
+/
+
+ACTIONX
+'A' /
+WWCT 'OPX'     > 0.75    AND /
+FPR < 100 /
+/
+
+WELPI
+  'PROD1' 1000 /
+/
+
+ENDACTIO
+
+TSTEP
+10 /
+
+        )"};
+
+    auto unit_system =  UnitSystem::newMETRIC();
+    const auto st = SummaryState{ std::chrono::system_clock::now() };
+    Schedule sched = make_schedule(deck_string);
+    const auto& action1 = sched.actions(0).get("A");
+    {
+        const auto& well = sched.getWell("PROD1", 0);
+        BOOST_CHECK_EQUAL( well.getWellPIScalingFactor(1.0), 1.0);
+    }
+
+    Action::Result action_result(true);
+    sched.applyAction(0, action1, action_result);
+    {
+        const auto& well = sched.getWell("PROD1", 0);
+        BOOST_CHECK( well.getWellPIScalingFactor(1.0) != 1.0);
```

Comment:
Ideally, this would also include a check that the scaling factor is (close to) its expected value, which of course is easier to do if you pass a value in the expected unit system:
```C++
const auto PI = unit_system.to_si(1.0, Unitsystem::measure::liquid_productivity_index);
const auto scaling = well.get.WellPIScalingFactor(PI);
BOOST_CHECK_CLOSE(scaling, 1000.0, 1.0e-10);
```